Mathlib/CategoryTheory/ChosenFiniteProducts.lean

```diff
@@ -547,4 +594,92 @@ noncomputable def monoidalOfChosenFiniteProducts : F.Monoidal :=
 
 end Functor
 
+namespace Functor.Monoidal
+
+variable {C : Type u} [Category.{v} C] [ChosenFiniteProducts C]
+  {D : Type u₁} [Category.{v₁} D] [ChosenFiniteProducts D] (F : C ⥤ D)
+
+section
+
+attribute [local instance] oplaxMonoidalOfChosenFiniteProducts
+
+@[reassoc (attr := simp)]
+lemma δ_fst {X Y : C} : OplaxMonoidal.δ F X Y ≫ fst _ _ = F.map (fst _ _) := by
+  simp [δ_of_chosenFiniteProducts]
+
+@[reassoc (attr := simp)]
+lemma δ_snd {X Y : C} : OplaxMonoidal.δ F X Y ≫ snd _ _ = F.map (snd _ _) := by
+  simp [δ_of_chosenFiniteProducts]
+
+@[reassoc (attr := simp)]
+lemma lift_δ {X Y Z : C} (f : X ⟶ Y) (g : X ⟶ Z) :
+    F.map (lift f g) ≫ OplaxMonoidal.δ F _ _ = lift (F.map f) (F.map g) := by
+  apply hom_ext <;> simp [← F.map_comp]
+
+end
+
+section
+
+open Limits
+
+variable [PreservesFiniteProducts F]
+
+attribute [local instance] monoidalOfChosenFiniteProducts
+
+@[reassoc (attr := simp)]
+lemma toUnit_ε {X : C} : toUnit (F.obj X) ≫ LaxMonoidal.ε F = F.map (toUnit X) :=
+  (cancel_mono (εIso _).inv).1 (toUnit_unique _ _)
+
+@[reassoc (attr := simp)]
+lemma lift_μ {X Y Z : C} (f : X ⟶ Y) (g : X ⟶ Z) :
+    lift (F.map f) (F.map g) ≫ LaxMonoidal.μ F _ _ = F.map (lift f g) :=
+  (cancel_mono (μIso _ _ _).inv).1 (by simp)
+
+@[reassoc (attr := simp)]
+lemma μ_fst {X Y : C} : LaxMonoidal.μ F X Y ≫ F.map (fst X Y) = fst (F.obj X) (F.obj Y) :=
+  (cancel_epi (μIso _ _ _).inv).1 (by simp)
+
+@[reassoc (attr := simp)]
+lemma μ_snd {X Y : C} : LaxMonoidal.μ F X Y ≫ F.map (snd X Y) = snd (F.obj X) (F.obj Y) :=
+  (cancel_epi (μIso _ _ _).inv).1 (by simp)
+
+
+end
+
+end Functor.Monoidal
+
+namespace Functor
+
+variable {C : Type u} [Category.{v} C] [ChosenFiniteProducts C]
+  {D : Type u₁} [Category.{v₁} D] [ChosenFiniteProducts D] (F : C ⥤ D)
+
+attribute [local instance] monoidalOfChosenFiniteProducts
+
+/-- A finite-product-preserving functor between categories with chosen finite products is
+braided. -/
+noncomputable def braidedOfChosenFiniteProducts [Limits.PreservesFiniteProducts F] : F.Braided :=
+  { monoidalOfChosenFiniteProducts F with
+    braided X Y := by
+      rw [← cancel_mono (Monoidal.μIso _ _ _).inv]
+      apply ChosenFiniteProducts.hom_ext <;> simp [← Functor.map_comp] }
+
+end Functor
+
+namespace NatTrans
+
+variable {C : Type u} [Category.{v} C] [ChosenFiniteProducts C]
+  {D : Type u₁} [Category.{v₁} D] [ChosenFiniteProducts D] (F G : C ⥤ D)
+  [Limits.PreservesFiniteLimits F] [Limits.PreservesFiniteLimits G]
+
+attribute [local instance] Functor.monoidalOfChosenFiniteProducts in
+theorem monoidal_of_preservesFiniteLimits (α : F ⟶ G) :
```

Comment:
Do not we need only the preservation of finite products here?

Reply:
Fixed, thanks.